test/blockmap.jl

```diff
@@ -136,7 +145,50 @@ using Test, LinearMaps, LinearAlgebra
             Lt = @inferred transform(L)
             @test Lt isa LinearMaps.LinearMap{elty}
             @test Lt * x ≈ transform(A) * x
-            @test Matrix(Lt) ≈ Matrix(transform(A))
+            @test Matrix(Lt) ≈ Matrix(transform(LinearMap(L))) ≈ Matrix(transform(A))
+            @test Matrix(transform(LinearMap(L))+transform(LinearMap(L))) ≈ 2Matrix(transform(A))
+            X = rand(elty, size(L, 1), 10)
+            Y = randn(elty, size(L, 2), 10)
+            for α in (0, 1, rand(elty)), β in (0, 1, rand(elty))
+                @test mul!(copy(Y), Lt, X, α, β) ≈ Y*β .+ transform(A)*X*α
+            end
+        end
+    end
+
+    @testset "block diagonal maps" begin
+        for elty in (Float64, ComplexF64)
+            m = 5; n = 6
+            M1 = 10*(1:m) .+ (1:(n+1))'; L1 = LinearMap(M1)
+            M2 = randn(elty, m, n+2); L2 = LinearMap(M2)
+            M3 = randn(elty, m, n+3); L3 = LinearMap(M3)
+
+            # Md = diag(M1, M2, M3, M2, M1) # unsupported so use sparse:
+            Md = Matrix(blockdiag(sparse.((M1, M2, M3, M2, M1))...))
+            x = randn(elty, size(Md, 2))
+            Bd = @inferred blockdiag(L1, L2, L3, L2, L1)
+            @test Matrix(@inferred blockdiag(L1)) == M1
+            @test Matrix(@inferred blockdiag(L1, L2)) == blockdiag(sparse.((M1, M2))...)
+            Bd2 = @inferred cat(L1, L2, L3, L2, L1; dims=(1,2))
```

Comment:
I should say that these tests are shamelessly stolen from @JeffFessler's PR #65. 😉